From 84f6d480d4b535d7d1dc018ee0718f4f41e7fc61 Mon Sep 17 00:00:00 2001
From: Quentin Lambert <lambert.quentin@gmail.com>
Date: Fri, 2 Sep 2016 11:48:53 +0200
Subject: [PATCH 055/198] crypto: ccp - add missing release in
 ccp_dmaengine_register

commit ba22a1e2aa8ef7f8467f755cfe44b79784febefe upstream.

ccp_dmaengine_register used to return with an error code before
releasing all resource. This patch adds a jump to the appropriate label
ensuring that the resources are properly released before returning.

This issue was found with Hector.

Signed-off-by: Quentin Lambert <lambert.quentin@gmail.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 drivers/crypto/ccp/ccp-dmaengine.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/ccp/ccp-dmaengine.c b/drivers/crypto/ccp/ccp-dmaengine.c
index 56b8aeb..020b461 100644
--- a/drivers/crypto/ccp/ccp-dmaengine.c
+++ b/drivers/crypto/ccp/ccp-dmaengine.c
@@ -654,8 +654,11 @@ int ccp_dmaengine_register(struct ccp_device *ccp)
 	dma_desc_cache_name = devm_kasprintf(ccp->dev, GFP_KERNEL,
 					     "%s-dmaengine-desc-cache",
 					     ccp->name);
-	if (!dma_desc_cache_name)
-		return -ENOMEM;
+	if (!dma_desc_cache_name){
+		ret = -ENOMEM;
+		goto err_cache;
+	}
+
 	ccp->dma_desc_cache = kmem_cache_create(dma_desc_cache_name,
 						sizeof(struct ccp_dma_desc),
 						sizeof(void *),
-- 
1.7.9.5

